modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxFinishFuture.java

```diff
@@ -485,6 +486,9 @@ private boolean finish(boolean commit,
             req.writeVersion(tx.writeVersion() != null ? tx.writeVersion() : tx.xidVersion());
 
             try {
+                if (isNull(cctx.discovery().node(n.id())))
+                    throw new ClusterTopologyCheckedException("Unable to send message (node left topology): " + n);
```

Comment:
i hope no need to throw exception here, "fut.onNodeLeft" would be enough.

Reply:
I think it's worth logging the problem.

Reply:
You're right, corrected.